core/shared/src/main/scala/fs2/Segment.scala

```diff
@@ -333,11 +333,12 @@ abstract class Segment[+O,+R] { self =>
 
       outerStep.map { outer =>
         step {
+          val innerRem = if (inner eq null) Segment.empty[O2].asInstanceOf[Segment[O2,S]] else inner.remainder
```

Comment:
I had a feeling since the result parameter is fixed to Unit for empty (since its just an alias to Chunk.empty).  So I guess Segment.empty won't work after all.